### ISSUE TITLE
HDBSCAN Topic Clustering

### ISSUE THREAD

**Issue by keyuchen21** [charlesdedampierre/BunkaTopics#60]
HDBSCAN has the ability to clustering the outliers into -1.
in this case with bunkatopic due to the topic id seems to be "arbitrary unique identifier"

in this case how do I know which topic is outliers?

**Comment by keyuchen21** [comment]
when I use 1000 example with HDBSCAN, the output is only 945?
<img width="1192" alt="Image" src="https://github.com/user-attachments/assets/ed0cdfbb-7af3-4d2c-b6ab-bc34835ade70" />

<img width="850" alt="Image" src="https://github.com/user-attachments/assets/e1b5e25b-6f73-4cf4-a82e-f939000c4af1" />

here is the code
`#!pip install bunkatopics
import datamapplot
from bunkatopics import Bunka
from datasets import load_dataset
import random
from sentence_transformers import SentenceTransformer
import umap

docs = load_dataset("bunkalab/medium-sample-technology")["train"]["title"][:1000]
len(docs)



embedding_model = SentenceTransformer(model_name_or_path="all-MiniLM-L6-v2")

# Load Projection Model

projection_model = umap.UMAP(
                n_components=2,
                random_state=42,
            )


# Initialize Bunka with your chosen model and language preference
bunka = Bunka(embedding_model=embedding_model,
              projection_model = projection_model)

# Fit Bunka to your text data
bunka.fit(docs)


from sklearn.cluster import KMeans
#clustering_model = KMeans(n_clusters=15)


from hdbscan import HDBSCAN

clustering_model = HDBSCAN(min_cluster_size=15, metric='euclidean', cluster_selection_method='eom', prediction_data=True)


df_topics =  bunka.get_topics(name_length=5, custom_clustering_model=clustering_model)# Specify the number of terms to describe each topic
df_topics


df_topics['size'].sum()

import pandas as pd
df_topics = pd.DataFrame([x.dict() for x in bunka.topics])
df_docs = pd.DataFrame([x.dict() for x in bunka.docs])
df_docs = pd.merge(df_docs, df_topics, on='topic_id', how='left')

labels = df_docs['name'].values.astype(str)
labels[:3]

doc_id = df_docs['doc_id'].values.astype(str)
content = df_docs['content'].values.astype(str)
len(data_map)
len(labels)`
